runtime/caching/cache.go

```diff
@@ -111,9 +116,14 @@ func (c *Cache) Require(ctx Context, options interface{}) (*Handle, error) {
 		c.entries = append(c.entries, entry)
 
 		go entry.created.Do(func() {
-			defer entry.ctx.dispose() // ensure resources are cleanup when constructor is done
+			defer entry.ctx.dispose()
+			// ensure resources are cleanup when constructor is done
+
+			c.monitor.Time("constructor-duration", func() {
+				entry.resource, entry.err = c.constructor(entry.ctx, options)
+			})
+			//metric to give creation time of constructor
```

Comment:
formatting... look how other comments have a space before them...

And maybe rephrase it...